src/it/scala/io/iohk/ethereum/sync/RegularSyncItSpec.scala

```diff
@@ -9,60 +9,77 @@ import org.scalatest.matchers.should.Matchers
+        }
+    }
+
+    "given a previously mined blockchain" in customTestCaseResourceM(FakePeer.start2FakePeersRes()) {
+      case (peer1, peer2) =>
+        val blockHeadersPerRequest = 200
+        for {
+          _ <- peer1.startRegularSync()
+          _ <- peer1.mineNewBlocks(100.milliseconds, blockHeadersPerRequest + 1)(IdentityUpdate)
+          _ <- peer1.waitForRegularSyncLoadLastBlock(blockHeadersPerRequest + 1)
+          _ <- peer2.startRegularSync()
+          _ <- peer2.connectToPeers(Set(peer1.node))
+          _ <- peer2.waitForRegularSyncLoadLastBlock(blockHeadersPerRequest + 1)
+        } yield {
+          assert(peer1.bl.getBestBlock().hash == peer2.bl.getBestBlock().hash)
+        }
+    }
   }
 
-  it should "sync blockchain progressing forward in the same time" in customTestCaseResourceM(
+  "peers should keep synced the same blockchain while their progressing forward" in customTestCaseResourceM(
     FakePeer.start2FakePeersRes()
   ) { case (peer1, peer2) =>
     val blockNumer: Int = 2000
     for {
-      _ <- peer2.startRegularSync().delayExecution(50.milliseconds)
-      _ <- peer2.importBlocksUntil(blockNumer)(IdentityUpdate)
-      _ <- peer1.connectToPeers(Set(peer2.node))
-      _ <- peer1.startRegularSync().delayExecution(500.milliseconds)
-      _ <- peer2.mineNewBlocks(2000.milliseconds, 2)(IdentityUpdate)
+      _ <- peer1.startRegularSync()
+      _ <- peer2.startRegularSync()
+      _ <- peer1.importBlocksUntil(blockNumer)(IdentityUpdate)
+      _ <- peer2.connectToPeers(Set(peer1.node))
+      _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumer)
+      _ <- peer2.mineNewBlocks(50.milliseconds, 2)(IdentityUpdate)
       _ <- peer1.waitForRegularSyncLoadLastBlock(blockNumer + 2)
+      _ <- peer1.mineNewBlocks(50.milliseconds, 2)(IdentityUpdate)
+      _ <- peer2.waitForRegularSyncLoadLastBlock(blockNumer + 4)
     } yield {
       assert(peer1.bl.getBestBlock().hash == peer2.bl.getBestBlock().hash)
     }
   }
 
-  it should "sync peers with divergent chains will be forced to resolve branches" in customTestCaseResourceM(
+  "peers with divergent chains will be forced to resolve branches" in customTestCaseResourceM(
     FakePeer.start2FakePeersRes()
   ) { case (peer1, peer2) =>
     val blockNumer: Int = 2000
     for {
-      _ <- peer2.importBlocksUntil(blockNumer)(IdentityUpdate)
-      _ <- peer2.startRegularSync().delayExecution(50.milliseconds)
+      _ <- peer1.startRegularSync()
+      _ <- peer2.startRegularSync()
       _ <- peer1.importBlocksUntil(blockNumer)(IdentityUpdate)
-      _ <- peer1.startRegularSync().delayExecution(50.milliseconds)
-      _ <- peer2.mineNewBlock(10)(IdentityUpdate).delayExecution(500.milliseconds)
-      _ <- peer2.mineNewBlock(10)(IdentityUpdate).delayExecution(500.milliseconds)
-      _ <- peer2.mineNewBlock(10)(IdentityUpdate).delayExecution(500.milliseconds)
+      _ <- peer2.importBlocksUntil(blockNumer)(IdentityUpdate)
```

Comment:
Isn't it prone to errors to have `importBlocksUntil` after starting the regular sync? That function goes directly into the db... I'm not sure how that could affect how syncing behaves